tck/server.ts

```diff
@@ -0,0 +1,132 @@
+import express from "express";
+import bodyParser from "body-parser";
+import {
+    JSONRPCParams,
+    JSONRPCRequest,
+    JSONRPCServer,
+    SimpleJSONRPCMethod,
+    createJSONRPCErrorResponse,
+} from "json-rpc-2.0";
+
+import * as methods from "./methods";
+import mapping from "./mapping";
+import { RpcMethodParams } from "./params/sdk";
+
+const server = new JSONRPCServer();
+
+// Helper function to register methods
+const registerMethods = (
+    methodsObject: Record<string, string>,
+    prefix = "",
+) => {
+    Object.entries(methodsObject).forEach(([methodName, method]) => {
+        if (methodName === "default" && typeof method === "object") {
+            // Recursively register methods inside the `default` property
+            registerMethods(method, prefix);
+        } else {
+            // Register the method with the correct prefix
+            const fullMethodName = prefix
+                ? `${prefix}.${methodName}`
+                : methodName;
+
+            server.addMethod(
+                fullMethodName,
+                method as unknown as SimpleJSONRPCMethod<void>,
+            );
+        }
+    });
+};
+
+// Register all methods, handling possible nested default export
+if ("default" in methods) {
+    registerMethods(methods.default);
+} else {
+    registerMethods(methods);
+}
+
+// Create mapping server method
+server.addMethod(
+    "mapping",
+    async (...args: [params: RpcMethodParams, serverParams?: void]) => {
+        // Basic mapping for unimplemented functions
+        return await mapping(args[0]);
+    },
+);
+
+/**
+ * Middleware to handle exceptions and customize error responses
+ */
+const exceptionMiddleware = async (
+    next: SimpleJSONRPCMethod,
+    request: JSONRPCRequest,
+    serverParams: JSONRPCParams,
+) => {
+    try {
+        return await next(request, serverParams);
+    } catch (error) {
+        if (error.status && error.status._code) {
+            return createJSONRPCErrorResponse(
```

Comment:
add some logging here, like in go `log.Printf("Error occurred processing JSON-RPC request: %s, Response error: %s", req, err)`.
You can also create different messages for hedera errors and internal errors.

Reply:
Also, please check how this func handles the `invalidParamError` because this error has it's own response code and should be handled separately from hedera error and internal error.